### PR TITLE
Presentation. Deprecated rule cleanup

### PR DESCRIPTION
**Pull request by karolisab**
Removed deprecated presentation rules.

Updated old json rule configurations to use the latest rule versions.

Updated documentation.

### CODE REVIEW

File: presentation/common/src/presentation-common/rules/content/modifiers/RelatedPropertiesSpecification.ts [hunk 1]
```diff
@@ -48,92 +46,11 @@ export enum RelatedPropertiesSpecialValues {
...
-   * properties are included if this attribute is not specified.
-   */
-  properties?: Array<string | PropertySpecification> | RelatedPropertiesSpecialValues;
-
-  /** Specifications for nested related properties */
-  nestedRelatedProperties?: RelatedPropertiesSpecification[];
-}
-
 /**
  * Sub-specification to include additional related instance properties.
```
Comment (grigasp) [inline presentation/common/src/presentation-common/rules/content/modifiers/RelatedPropertiesSpecification.ts:50]: Missing doc link
```
 * @see [More details]($docs/learning/presentation/Content/RelatedPropertiesSpecification.md)
```

File: presentation/common/src/presentation-common/rules/content/modifiers/RelatedPropertiesSpecification.ts [hunk 2]
```diff
@@ -144,6 +61,7 @@ export interface RelatedPropertiesSpecificationNew {
    * polymorphically. This means properties of not only the target class, but also all its subclasses
    * are loaded.
    *
+   *
```
Comment (grigasp) [inline presentation/common/src/presentation-common/rules/content/modifiers/RelatedPropertiesSpecification.ts:64]: unnecessary line

File: full-stack-tests/presentation/src/frontend/Hierarchies.test.ts [hunk 3]
```diff
@@ -2,16 +2,23 @@
 * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
 * See LICENSE.md in the project root for license terms and full copyright notice.
 *--------------------------------------------------------------------------------------------*/
-import { expect } from "chai";
+import {expect} from "chai";
```
Comment (grigasp) [inline full-stack-tests/presentation/src/frontend/Hierarchies.test.ts:5]: formatting

File: presentation/backend/assets/supplemental-presentation-rules/BisCore.PresentationRuleSet.json [hunk 4]
```diff
@@ -227,145 +227,120 @@
...
-            "schemaName": "BisCore",
-            "classNames": [
-              "ElementUniqueAspect"
-            ]
+          "propertiesSource": {
+            "relationship": {
+              "className": "BisCore",
+              "schemaName": "ElementOwnsUniqueAspect"
+            },
+            "targetClass": {
```
Comment (grigasp) [inline presentation/backend/assets/supplemental-presentation-rules/BisCore.PresentationRuleSet.json:235]: unnecessary target class - it matches target of the relationship

File: presentation/backend/assets/supplemental-presentation-rules/BisCore.PresentationRuleSet.json [hunk 5]
```diff
@@ -227,145 +227,120 @@
...
-            "schemaName": "BisCore",
-            "classNames": [
-              "ElementMultiAspect"
-            ]
+          "propertiesSource": {
+            "relationship": {
+              "schemaName": "BisCore",
+              "className": "ElementOwnsMultiAspects"
+            },
+            "targetClass": {
```
Comment (grigasp) [inline presentation/backend/assets/supplemental-presentation-rules/BisCore.PresentationRuleSet.json:251]: unnecessary target class - it matches target of the relationship

File: presentation/backend/assets/supplemental-presentation-rules/BisCore.PresentationRuleSet.json [hunk 6]
```diff
@@ -227,145 +227,120 @@
...
-            "schemaName": "BisCore",
-            "classNames": [
-              "LinkElement"
-            ]
+          "propertiesSource": {
+            "relationship": {
+              "schemaName": "BisCore",
+              "className": "ElementHasLinks"
+            },
+            "targetClass": {
```
Comment (grigasp) [inline presentation/backend/assets/supplemental-presentation-rules/BisCore.PresentationRuleSet.json:267]: unnecessary target class - it matches target of the relationship

File: presentation/backend/assets/supplemental-presentation-rules/BisCore.PresentationRuleSet.json [hunk 7]
```diff
@@ -227,145 +227,120 @@
...
-                "schemaName": "BisCore",
-                "classNames": [
-                  "LinkElement"
-                ]
+              "propertiesSource": {
+                "relationship": {
+                  "schemaName": "BisCore",
+                  "className": "ElementHasLinks"
+                },
+                "targetClass": {
```
Comment (grigasp) [inline presentation/backend/assets/supplemental-presentation-rules/BisCore.PresentationRuleSet.json:298]: unnecessary target class - it matches target of the relationship

File: presentation/backend/assets/supplemental-presentation-rules/BisCore.PresentationRuleSet.json [hunk 8]
```diff
@@ -663,20 +638,18 @@
...
-            "schemaName": "BisCore",
-            "classNames": [
-              "Element"
-            ]
+          "propertiesSource": {
+            "relationship": {
+              "schemaName": "BisCore",
+              "className": "DrawingGraphicRepresentsElement"
+            },
+            "targetClass": {
```
Comment (grigasp) [inline presentation/backend/assets/supplemental-presentation-rules/BisCore.PresentationRuleSet.json:646]: unnecessary target class - it matches target of the relationship

File: presentation/backend/assets/supplemental-presentation-rules/BisCore.PresentationRuleSet.json [hunk 9]
```diff
@@ -690,20 +663,18 @@
...
-            "schemaName": "BisCore",
-            "classNames": [
-              "Element"
-            ]
+          "propertiesSource": {
+            "relationship": {
+              "schemaName": "BisCore",
+              "className": "GraphicalElement3dRepresentsElement"
+            },
+            "targetClass": {
```
Comment (grigasp) [inline presentation/backend/assets/supplemental-presentation-rules/BisCore.PresentationRuleSet.json:671]: unnecessary target class - it matches target of the relationship

File: test-apps/presentation-test-app/assets/presentation_rules/IModelIndex.PresentationRuleSet.json [hunk 10]
```diff
@@ -36,39 +36,39 @@
...
-              "Subject"
-            ]
-          },
+          "relationshipPaths": [
+            {
+              "relationship": {
+                "schemaName": "BisCore",
+                "className": "SubjectOwnsSubjects"
+              },
+              "targetClass": {
```
Comment (grigasp) [inline test-apps/presentation-test-app/assets/presentation_rules/IModelIndex.PresentationRuleSet.json:45]: unnecessary target class - it matches target of the relationship

File: test-apps/presentation-test-app/assets/presentation_rules/IModelIndex.PresentationRuleSet.json [hunk 11]
```diff
@@ -36,39 +36,39 @@
...
```
Comment (grigasp) [inline test-apps/presentation-test-app/assets/presentation_rules/IModelIndex.PresentationRuleSet.json:65]: unnecessary target class - it matches target of the relationship

File: test-apps/presentation-test-app/assets/presentation_rules/VisibilityWidget.SpatialBreakdown-GroupedByClass.PresentationRuleSet.json [hunk 12]
```diff
@@ -1,89 +1,91 @@
...
+          ],
+          "arePolymorphic": true,
+          "relatedInstances": [
+            {
+              "relationshipPath": {
+                "relationship": {
+                  "schemaName": "BisCore",
+                  "className": "ModelContainsElements"
+                },
+                "targetClass": {
```
Comment (grigasp) [inline test-apps/presentation-test-app/assets/presentation_rules/VisibilityWidget.SpatialBreakdown-GroupedByClass.PresentationRuleSet.json:26]: unnecessary target class - it matches target of the relationship

File: test-apps/presentation-test-app/assets/presentation_rules/VisibilityWidget.SpatialBreakdown-GroupedByClass.PresentationRuleSet.json [hunk 13]
```diff
@@ -1,89 +1,91 @@
...
+      "specifications": [
+        {
+          "specType": "RelatedInstanceNodes",
+          "relationshipPaths": [
+            {
+              "relationship": {
+                "schemaName": "SpatialComposition",
+                "className": "CompositeComposesSubComposites"
+              },
+              "targetClass": {
```
Comment (grigasp) [inline test-apps/presentation-test-app/assets/presentation_rules/VisibilityWidget.SpatialBreakdown-GroupedByClass.PresentationRuleSet.json:54]: unnecessary target class - it matches target of the relationship

File: test-apps/presentation-test-app/assets/presentation_rules/VisibilityWidget.SpatialBreakdown-GroupedByClass.PresentationRuleSet.json [hunk 14]
```diff
@@ -1,89 +1,91 @@
...
+      "specifications": [
+        {
+          "specType": "RelatedInstanceNodes",
+          "relationshipPaths": [
+            {
+              "relationship": {
+                "schemaName": "SpatialComposition",
+                "className": "CompositeOverlapsSpatialElements"
+              },
+              "targetClass": {
```
Comment (grigasp) [inline test-apps/presentation-test-app/assets/presentation_rules/VisibilityWidget.SpatialBreakdown-GroupedByClass.PresentationRuleSet.json:78]: unnecessary target class - it matches target of the relationship

File: test-apps/presentation-test-app/assets/presentation_rules/VisibilityWidget.SpatialBreakdown.PresentationRuleSet.json [hunk 15]
```diff
@@ -1,89 +1,91 @@
...
+          ],
+          "arePolymorphic": true,
+          "relatedInstances": [
+            {
+              "relationshipPath": {
+                "relationship": {
+                  "schemaName": "BisCore",
+                  "className": "ModelContainsElements"
+                },
+                "targetClass": {
```
Comment (grigasp) [inline test-apps/presentation-test-app/assets/presentation_rules/VisibilityWidget.SpatialBreakdown.PresentationRuleSet.json:26]: unnecessary target class - it matches target of the relationship

File: test-apps/presentation-test-app/assets/presentation_rules/VisibilityWidget.SpatialBreakdown.PresentationRuleSet.json [hunk 16]
```diff
@@ -1,89 +1,91 @@
...
+      "specifications": [
+        {
+          "specType": "RelatedInstanceNodes",
+          "relationshipPaths": [
+            {
+              "relationship": {
+                "schemaName": "SpatialComposition",
+                "className": "CompositeComposesSubComposites"
+              },
+              "targetClass": {
```
Comment (grigasp) [inline test-apps/presentation-test-app/assets/presentation_rules/VisibilityWidget.SpatialBreakdown.PresentationRuleSet.json:54]: unnecessary target class - it matches target of the relationship

File: test-apps/presentation-test-app/assets/presentation_rules/VisibilityWidget.SpatialBreakdown.PresentationRuleSet.json [hunk 17]
```diff
@@ -1,89 +1,91 @@
...
+      "specifications": [
+        {
+          "specType": "RelatedInstanceNodes",
+          "relationshipPaths": [
+            {
+              "relationship": {
+                "schemaName": "SpatialComposition",
+                "className": "CompositeOverlapsSpatialElements"
+              },
+              "targetClass": {
```
Comment (grigasp) [inline test-apps/presentation-test-app/assets/presentation_rules/VisibilityWidget.SpatialBreakdown.PresentationRuleSet.json:78]: unnecessary target class - it matches target of the relationship

File: ui/framework/rulesets/Categories.json [hunk 18]
```diff
@@ -53,19 +53,19 @@
...
-            ]
-          },
-          "requiredDirection": "Forward",
+          "relationshipPaths": [
+            {
+              "relationship": {
+                "schemaName": "BisCore",
+                "className": "CategoryOwnsSubCategories"
+              },
+              "targetClass": {
```
Comment (grigasp) [inline ui/framework/rulesets/Categories.json:62]: unnecessary target class - it matches target of the relationship

File: ui/framework/rulesets/Models.json [hunk 19]
```diff
@@ -21,11 +21,13 @@
           "groupByLabel": false,
           "relatedInstances": [
             {
-              "relationship": {
-                "schemaName": "BisCore",
-                "className": "ModelModelsElement"
+              "relationshipPath": {
+                "relationship": {
+                  "schemaName": "BisCore",
+                  "className": "ModelContainsElements"
```
Comment (grigasp) [inline ui/framework/rulesets/Models.json:27]: ModelModelsElement?

File: ui/framework/src/ui-framework/imodel-components/spatial-tree/SpatialBreakdown.json [hunk 20]
```diff
@@ -1,89 +1,91 @@
...
+          ],
+          "arePolymorphic": true,
+          "relatedInstances": [
+            {
+              "relationshipPath": {
+                "relationship": {
+                  "schemaName": "BisCore",
+                  "className": "ModelContainsElements"
+                },
+                "targetClass": {
```
Comment (grigasp) [inline ui/framework/src/ui-framework/imodel-components/spatial-tree/SpatialBreakdown.json:26]: unnecessary target class - it matches target of the relationship

File: ui/framework/src/ui-framework/imodel-components/spatial-tree/SpatialBreakdown.json [hunk 21]
```diff
@@ -1,89 +1,91 @@
...
+      "specifications": [
+        {
+          "specType": "RelatedInstanceNodes",
+          "relationshipPaths": [
+            {
+              "relationship": {
+                "schemaName": "SpatialComposition",
+                "className": "CompositeComposesSubComposites"
+              },
+              "targetClass": {
```
Comment (grigasp) [inline ui/framework/src/ui-framework/imodel-components/spatial-tree/SpatialBreakdown.json:54]: unnecessary target class - it matches target of the relationship

File: ui/framework/src/ui-framework/imodel-components/spatial-tree/SpatialBreakdown.json [hunk 22]
```diff
@@ -1,89 +1,91 @@
...
+      "specifications": [
+        {
+          "specType": "RelatedInstanceNodes",
+          "relationshipPaths": [
+            {
+              "relationship": {
+                "schemaName": "SpatialComposition",
+                "className": "CompositeOverlapsSpatialElements"
+              },
+              "targetClass": {
```
Comment (grigasp) [inline ui/framework/src/ui-framework/imodel-components/spatial-tree/SpatialBreakdown.json:78]: unnecessary target class - it matches target of the relationship

File: ui/framework/src/ui-framework/imodel-components/spatial-tree/SpatialBreakdown.GroupedByClass.json [hunk 23]
```diff
@@ -1,89 +1,91 @@
...
+          ],
+          "arePolymorphic": true,
+          "relatedInstances": [
+            {
+              "relationshipPath": {
+                "relationship": {
+                  "schemaName": "BisCore",
+                  "className": "ModelContainsElements"
+                },
+                "targetClass": {
```
Comment (grigasp) [inline ui/framework/src/ui-framework/imodel-components/spatial-tree/SpatialBreakdown.GroupedByClass.json:26]: unnecessary target class - it matches target of the relationship

File: ui/framework/src/ui-framework/imodel-components/spatial-tree/SpatialBreakdown.GroupedByClass.json [hunk 24]
```diff
@@ -1,89 +1,91 @@
...
+      "specifications": [
+        {
+          "specType": "RelatedInstanceNodes",
+          "relationshipPaths": [
+            {
+              "relationship": {
+                "schemaName": "SpatialComposition",
+                "className": "CompositeComposesSubComposites"
+              },
+              "targetClass": {
```
Comment (grigasp) [inline ui/framework/src/ui-framework/imodel-components/spatial-tree/SpatialBreakdown.GroupedByClass.json:54]: unnecessary target class - it matches target of the relationship

File: ui/framework/src/ui-framework/imodel-components/spatial-tree/SpatialBreakdown.GroupedByClass.json [hunk 25]
```diff
@@ -1,89 +1,91 @@
...
+      "specifications": [
+        {
+          "specType": "RelatedInstanceNodes",
+          "relationshipPaths": [
+            {
+              "relationship": {
+                "schemaName": "SpatialComposition",
+                "className": "CompositeOverlapsSpatialElements"
+              },
+              "targetClass": {
```
Comment (grigasp) [inline ui/framework/src/ui-framework/imodel-components/spatial-tree/SpatialBreakdown.GroupedByClass.json:78]: unnecessary target class - it matches target of the relationship

**Review by grigasp — CHANGES_REQUESTED** [review]

**Review by grigasp — COMMENTED** [review]
Looks good. Missing pieces:
- [ ] `rush extract-api`
- [ ] `rush change`
- [ ] `NextVersion.md`

**Review by grigasp — APPROVED** [review]

**Review by bsteinbk — APPROVED** [review]

**Pull request merged**